src/root_readspeed.cxx

```diff
@@ -45,16 +51,48 @@ Args ParseArgs(int argc, char **argv)
    unsigned int nThreads = 0;
 
    enum class EArgState { kNone, kTrees, kFiles, kBranches, kThreads } argState = EArgState::kNone;
+   enum class EBranchState { kNone, kRegular, kRegex, kAll } branchState = EBranchState::kNone;
+
    for (int i = 1; i < argc; ++i) {
-      if (std::strcmp(argv[i], "--trees") == 0)
+      std::string arg(argv[i]);
+
+      if (arg.compare("--trees") == 0) {
          argState = EArgState::kTrees;
-      else if (std::strcmp(argv[i], "--files") == 0)
+      } else if (arg.compare("--files") == 0) {
          argState = EArgState::kFiles;
-      else if (std::strcmp(argv[i], "--branches") == 0)
+      } else if (arg.compare("--all-branches") == 0) {
+         argState = EArgState::kNone;
+         if (branchState != EBranchState::kNone && branchState != EBranchState::kAll) {
+            std::cerr << "Options --all-branches, --branches, and --branches-regex are mutually exclusive. You can use "
+                         "only one."
+                      << std::endl;
+            return {};
+         }
+         branchState = EBranchState::kAll;
+         d.fUseRegex = true;
+         d.fBranchNames = {".*"};
+      } else if (arg.compare("--branches") == 0) {
+         argState = EArgState::kBranches;
+         if (branchState != EBranchState::kNone && branchState != EBranchState::kRegular) {
+            std::cerr << "Options --all-branches, --branches, and --branches-regex are mutually exclusive. You can use "
+                         "only one."
+                      << std::endl;
+            return {};
+         }
+         branchState = EBranchState::kRegular;
+      } else if (arg.compare("--branches-regex") == 0) {
          argState = EArgState::kBranches;
-      else if (std::strcmp(argv[i], "--threads") == 0)
+         if (branchState != EBranchState::kNone && branchState != EBranchState::kRegex) {
+            std::cerr << "Options --all-branches, --branches, and --branches-regex are mutually exclusive. You can use "
+                         "only one."
+                      << std::endl;
```

Comment:
could do:

```suggestion
            std::cerr << branchOptionsErrMsg;
```

after lifting the message into a std::string to avoid repeating the string three times.